planning/MartenV4.md

```diff
@@ -0,0 +1,114 @@
+# Marten v4 Ideas
+
+## Pulling it Off
+
+We've got the typical problems of needing to address incoming pull requests and bug issues in master while probably needing to have a long lived branch for *v4*. 
+
+Here are some thoughts:
+
+* First do a bug sweep 3.12 release to address as many of the tactical problems as we can before branching to v4?
+* Start with the unit testing improvements as a way to speed up the build before we dive into much more of this? 
```

Comment:
@jeremydmiller could you elabortate on that part?

Reply:
I think that if we could get aggressive w/ xUnit ClassFixtures to share DocumentStore objects and schemas, we could speed up the testing feedback. Maybe even allow a little multithreading w/ different schemas too.

I'm worried a little bit about the cold start overhead on the testing by re-introducing the runtime codegen, but it would make all the development go faster in the longer run by speeding up the test times.

Especially now that we're doing so much matrix testing

Reply:
Ok, understood, so that's mostly what's described below. I agree with that 👍

Reply:
I think this would be the very first thing I'd want to tackle. And maybe even immediately as a helper for v3.12 even.